packages/vscode-extension/src/commonlib/telemetry.ts

```diff
@@ -155,4 +155,8 @@ export class VSCodeTelemetryReporter extends vscode.Disposable implements Teleme
       this.reporter.sendTelemetryException(error, properties, measurements);
     }
   }
+
+  async dispose() {
```

Comment:
Is this duplicated with the constructor? Line 40 already added dispose callback in `super()`;

Reply:
Since dispose callback has already been added in `super()`, can I just call `this.dispose()` instead of adding a new one?

Reply:
Do you mean in this dispose function call 'await this.dispose()' instead of 'await this.reporter.dispose()'?

Reply:
I mean not to add a `dispose()` function, just calling `vscodeTelemetryReporter.dispose()` when needed. Since its constructor already added dispose callback to `super()`. So calling `vscodeTelemetryReporter.dispose()` is actually calling `vscode.Disposable.dispose()`.

Reply:
Just to confirm with you: will vscode.Disposable.dispose() call the callOnDispose function with await if it's async function?

Reply:
![image](https://user-images.githubusercontent.com/73154171/149471572-6fb902a5-a7d8-4d55-9397-e8016e3ac8f1.png)
I'm afraid by default vscode.Disposable.dispose() will not await the function

Reply:
Oh, that may be the cause of missing telemetry - vscode only triggers the async dispose, but not wait for it.

Reply:
Hope so~

Reply:
One more concern - will the added `VSCodeTelemetryReporter.dispose()` override `vscode.Disposable.dispose()`?

Reply:
Just to keep code clean / remove unused code : )